src/api/system.c

```diff
@@ -1150,6 +1150,24 @@ static int f_text_input(lua_State* L) {
   return 0;
 }
 
+static int f_setenv(lua_State* L) {
+  size_t keylen, vallen;
+  const char *key = luaL_checklstring(L, 1, &keylen);
+  const char *val = luaL_checklstring(L, 2, &vallen);
```

Comment:
If we aren't using the length, we can avoid querying for it.
```suggestion
  const char *key = luaL_checkstring(L, 1);
  const char *val = luaL_checkstring(L, 2);
```